### PR TITLE
fold: move to clap, add tests

### PR DESCRIPTION
**Pull request by ycd**
This PR includes these changes for fold: 

* Moved to clap
* Added tests and fixtures for testing compatibility against obsolote syntax

### CODE REVIEW

File: src/uu/fold/src/fold.rs [hunk 1]
```diff
@@ -10,58 +10,82 @@
...
-    } else {
-        obs_width
+        .arg(Arg::with_name(options::FILE).hidden(true).multiple(true))
+        .get_matches_from(args.clone());
+
+    let bytes = matches.is_present(options::BYTES);
+    let spaces = matches.is_present(options::SPACES);
+    let poss_width = match matches.is_present(options::WIDTH) {
+        true => matches.value_of(options::WIDTH).map(|v| v.to_owned()),
+        false => obs_width,
```
Comment (tertsdiepraam) [inline src/uu/fold/src/fold.rs:73]: Could be done like this, which seems slightly more readable:
```
let poss_width = match matches.value_of(options::WIDTH) {
    Some(v) => Some(v.to_owned()),
    None => obs_width,
};
```

**Review by tertsdiepraam — COMMENTED** [review]
Nice job! I found only two very minor things to change, otherwise I think you got all the `clap` configuration exactly right.

File: tests/by-util/test_fold.rs [hunk 2]
```diff
@@ -57,3 +57,13 @@ fn test_word_boundary_split_should_preserve_empty_lines() {
         .succeeds()
         .stdout_is("0\n1\n\n2\n\n\n");
 }
+
+#[test]
+fn test_obsolote_syntax() {
```
[the remark this answers is not in the document]
Comment (ycd) [reply tests/by-util/test_fold.rs]: i thought you didn't understand what obsolete syntax is and i was writing a explanatory comment about it, then i understand i made a typo LOL. :(

File: src/uu/fold/src/fold.rs (hunk 1 above)
Comment (ycd) [reply src/uu/fold/src/fold.rs]: sure, updated :)

File: src/uu/fold/Cargo.toml [hunk 3]
```diff
@@ -15,7 +15,8 @@ edition = "2018"
 path = "src/fold.rs"
 
 [dependencies]
-uucore = { version=">=0.0.8", package="uucore", path="../../uucore" }
+clap = "2.33"
+uucore = { version=">=0.0.7", package="uucore", path="../../uucore" }
```
Comment (sylvestre) [inline src/uu/fold/Cargo.toml:19]:
```suggestion
uucore = { version=">=0.0.8", package="uucore", path="../../uucore" }
```

**Comment by sylvestre** [comment]
Some tests failure:
```

run: D:\a\coreutils\coreutils\target\i686-pc-windows-gnu\debug\coreutils.exe fold -5 -s space_separated_words.txt
thread 'test_fold::test_obsolete_syntax' panicked at 'assertion failed: `(left == right)`
  left: `"test1\n \ntest2\n \ntest3\n \ntest4\n \ntest5\n \ntest6\n "`,
 right: `"test1\n \ntest2\n \ntest3\n \ntest4\n \ntest5\n \ntest6\n \n"`', tests\common\util.rs:190:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Pull request merged**